lib/src/react_client/synthetic_event_wrappers.dart

```diff
@@ -4,153 +4,621 @@
+///
+/// See: <https://facebook.github.io/react/docs/events.html#syntheticevent>
 @JS()
-@anonymous
 class SyntheticEvent {
+  /// [SyntheticEvent]s cannot be manually instantiated.
+  ///
+  /// Use `createSyntheticEvent` instead.
+  external factory SyntheticEvent._();
+
+  /// Indicates whether the [Event] bubbles up through the DOM or not.
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/API/Event/bubbles>
   external bool get bubbles;
+
+  /// Indicates whether the [Event] is cancelable or not.
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/API/Event/cancelable>
   external bool get cancelable;
+
+  /// Identifies the current target for the event, as the [Event] traverses the DOM.
+  ///
+  /// It always refers to the [Element] the [Event] handler has been attached to as opposed to [target] which identifies
+  /// the [Element] on which the [Event] occurred.
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget>
   external get currentTarget;
+
+  /// Indicates whether or not [preventDefault] was called on the event.
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/API/Event/defaultPrevented>
   external bool get defaultPrevented;
+
+  /// Indicates which phase of the [Event] flow is currently being evaluated.
+  ///
+  /// Possible values:
+  ///
+  /// > [Event.CAPTURING_PHASE] (1) - The [Event] is being propagated through the [target]'s ancestor objects. This
+  /// process starts with the Window, then [HtmlDocument], then the [HtmlHtmlElement], and so on through the [Element]s
+  /// until the [target]'s parent is reached. Event listeners registered for capture mode when
+  /// [EventTarget.addEventListener] was called are triggered during this phase.
+  ///
+  /// > [Event.AT_TARGET] (2) - The [Event] has arrived at the [target]. Event listeners registered for this phase are
+  /// called at this time. If [bubbles] is `false`, processing the [Event] is finished after this phase is complete.
+  ///
+  /// > [Event.BUBBLING_PHASE] (3) - The [Event] is propagating back up through the [target]'s ancestors in reverse
+  /// order, starting with the parent, and eventually reaching the containing Window. This is known as bubbling, and
+  /// occurs only if [bubbles] is `true`. [Event] listeners registered for this phase are triggered during this process.
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/API/Event/eventPhase>
   external num get eventPhase;
+
+  /// Is `true` when the [Event] was generated by a user action, and `false` when the [Event] was created or modified
+  /// by a script or dispatched via [Event.dispatchEvent].
+  ///
+  /// __Read Only__
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/API/Event/isTrusted>
   external bool get isTrusted;
-  external get nativeEvent;
-  external get target;
+
+  /// Native browser event that [SyntheticEvent] wraps around.
```

Comment:
#nit Referencing  SyntheticEvent isn't helpful here; instead, reference "this"

- https://dart.dev/guides/language/effective-dart/documentation#avoid-redundancy-with-the-surrounding-context
- https://dart.dev/guides/language/effective-dart/documentation#prefer-using-this-instead-of-the-to-refer-to-a-members-instance
```suggestion
  /// The native browser event this wraps.
```